spec/helpers/application_helper/buttons/vm_publish_spec.rb

```diff
@@ -3,27 +3,24 @@
     it_behaves_like "when record is orphaned"
     it_behaves_like "when record is archived"
 
-    context "when record is not cloneable and vendor is redhat" do
-      before do
-        @record = FactoryGirl.create(:vm_redhat)
-      end
+    context "when vendor is redhat" do
+      let(:ems) { FactoryGirl.create(:ems_redhat) }
+      let(:vm) { FactoryGirl.create(:vm_redhat, :ext_management_system => ems) }
 
-      it "will be skipped" do
+      it "returns false" do
         view_context = setup_view_context_with_sandbox({})
-        button = described_class.new(view_context, {}, {'record' => @record}, {})
+        button = described_class.new(view_context, {}, {:record => vm}, {})
         expect(button.visible?).to be_falsey
       end
     end
```

Comment:
same with the button, if we can remove the button class, we could also remove this spec